model2vec/model.py

```diff
@@ -465,9 +465,14 @@ def load_local(cls: type[StaticModel], path: PathLike) -> StaticModel:
         :return: A StaticModel
         :raises: ValueError if the path is not a directory.
         """
-        path = Path(path)
-        if not path.is_dir():
-            raise ValueError(f"Path {path} is not a directory.")
+        if isinstance(path, str):
+            path = Path(path)
+
+        if isinstance(path, Path):
+            # Only check if we're sure this is a path.
+            # It could be a cloudpathlib path, or something else.
```

Comment:
In this `if` statement `path` is always a `Path`. I.e., `AnyPath` doesn't have `Path` as base class, so this comment isn't true